### PR TITLE
Add Lombok sample and Apache logging hints

### PR DESCRIPTION
**Pull request by joshlong**
adds support for Lombok's `@Log4j2` synthetic logs

### CODE REVIEW

**Comment by sdeleuze** [comment]
Hi Josh, thanks for your contribution.

I think a sample will be required for that one, especially given the fact that Log4j2 has known issues on native and is currently not supported, see #115.

Please add a `commandlineruner-log4j2` sample with Log4j2 + Lombok to this PR in order to allow us to evaluate how far we are for getting that working.

**Comment by joshlong** [comment]
Hi there, I've updated the PR with a `lombok-log4j2` sample. Thank you for your consideration.

File: spring-aot/src/main/java/org/springframework/nativex/type/TypeSystem.java [hunk 1]
```diff
@@ -311,7 +323,8 @@ public Type resolve(org.objectweb.asm.Type type, boolean silent) {
 	public Type Lresolve(String desc, boolean silent) {
```
Comment (lyndseyBeil) [inline spring-aot/src/main/java/org/springframework/nativex/type/TypeSystem.java:323]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Comment by sdeleuze** [comment]
@joshlong Thanks for the sample. I think we need to clarify what is the scope here, as pointed out in my previous comment, as far as I know Log4j2 is not supported in Spring Native (cf. #115) and the sample provided is using Logback underneath not Log4j2. I guess what happens in the provided use case is that `@Log4j2` generates code using Apache Commons Logging APIs provided by Spring. Is it your understanding as well?

**Comment by sdeleuze** [comment]
Maybe `@Slf4j` would be a better fit here? Did you try it?

**Comment by joshlong** [comment]
I'll try that out

**Comment by joshlong** [comment]
yep, looks like switching out `@Slf4j` works too

**Comment by sdeleuze** [comment]
I merged a reworked version, focusing the sample on Lombok but the hint on Apache Logging. Thanks for your contribution.

**Pull request closed without merging**